contracts/test/integration/L2PlusSubnet.t.sol

```diff
@@ -0,0 +1,857 @@
+import {GatewayFacetsHelper} from "../helpers/GatewayFacetsHelper.sol";
+import {SubnetActorFacetsHelper} from "../helpers/SubnetActorFacetsHelper.sol";
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ERC20PresetFixedSupply} from "../helpers/ERC20PresetFixedSupply.sol";
+
+import "forge-std/console.sol";
+
+contract L2PlusSubnetTest is Test, IntegrationTestBase {
+    using SubnetIDHelper for SubnetID;
+    using CrossMsgHelper for IpcEnvelope;
+    using GatewayFacetsHelper for GatewayDiamond;
+    using SubnetActorFacetsHelper for SubnetActorDiamond;
+    using AssetHelper for Asset;
+
+    RootSubnetDefinition public rootNetwork;
+    // native subnets
+    TestSubnetDefinition public nativeSubnet;
+    TestSubnetDefinition[] public nativeL3Subnets;
+
+    // token subnets
+    IERC20 public token;
+    TestSubnetDefinition public tokenSubnet;
+    TestSubnetDefinition[] public nativeL3SubnetsWithTokenParent;
+    IERC20 public tokenL3;
+    TestSubnetDefinition[] public tokenL3SubnetsWithTokenParent;
+
+    function setUp() public override {
+        SubnetID memory rootNetworkName = SubnetID({root: ROOTNET_CHAINID, route: new address[](0)});
+        require(rootNetworkName.isRoot(), "not root");
+
+        GatewayDiamond rootGateway = createGatewayDiamond(gatewayParams(rootNetworkName));
+
+        rootNetwork = RootSubnetDefinition({
+            gateway: rootGateway,
+            gatewayAddr: address(rootGateway),
+            id: rootNetworkName
+        });
+
+        address[] memory nativeSubnetPath;
+        nativeSubnet = createNativeSubnet(nativeSubnetPath, rootNetwork.gatewayAddr, rootNetwork.id);
+
+        nativeL3Subnets.push(createNativeSubnet(nativeSubnet.path, nativeSubnet.gatewayAddr, nativeSubnet.id));
+        nativeL3Subnets.push(createNativeSubnet(nativeSubnet.path, nativeSubnet.gatewayAddr, nativeSubnet.id));
+
+        address[] memory tokenSubnetPath;
+        token = new ERC20PresetFixedSupply("TestToken", "TEST", 1_000_000, address(this));
+        tokenSubnet = createTokenSubnet(address(token), tokenSubnetPath, rootNetwork.gatewayAddr, rootNetworkName);
+
+        nativeL3SubnetsWithTokenParent.push(
+            createNativeSubnet(tokenSubnet.path, tokenSubnet.gatewayAddr, tokenSubnet.id)
+        );
+        nativeL3SubnetsWithTokenParent.push(
+            createNativeSubnet(tokenSubnet.path, tokenSubnet.gatewayAddr, tokenSubnet.id)
+        );
+
+        tokenL3 = new ERC20PresetFixedSupply("TestL3Token", "TEST3", 1_000_000, address(this));
+
+        tokenL3SubnetsWithTokenParent.push(
+            createTokenSubnet(address(tokenL3), tokenSubnet.path, tokenSubnet.gatewayAddr, tokenSubnet.id)
+        );
+        tokenL3SubnetsWithTokenParent.push(
+            createTokenSubnet(address(tokenL3), tokenSubnet.path, tokenSubnet.gatewayAddr, tokenSubnet.id)
+        );
+
+        printActors();
+    }
+
+    function createNativeSubnet(
+        address[] memory subnetPath,
```

Comment:
shouldnt this be parent subnet id? How do you know the subnet path before it's created?